be/src/exprs/vectorized/math_functions.cpp

```diff
@@ -441,20 +441,23 @@ ColumnPtr MathFunctions::conv_string(FunctionContext* context, const starrocks::
         }
 
         StringParser::ParseResult parse_res;
-        int64_t decimal_num = StringParser::string_to_int<int64_t>(reinterpret_cast<char*>(string_value.data),
+        int128_t decimal_num = StringParser::string_to_int<int128_t>(reinterpret_cast<char*>(string_value.data),
                                                                    string_value.size, src_base_value, &parse_res);
-
         if (src_base_value < 0 && decimal_num >= 0) {
             result.append_null();
             continue;
         }
-
-        if (!handle_parse_result(dest_base_value, &decimal_num, parse_res)) {
+        int64_t decimal64_num = static_cast<int64_t>(decimal_num);
+        // recheck overflow
+        if (parse_res == StringParser::PARSE_SUCCESS && decimal_num > std::numeric_limits<uint64_t>::max()) {
```

Comment:
Why use `std::numeric_limits<uint64_t>::max()` but not `std::numeric_limits<int64_t>::max()`

Reply:
According to the description of [mysql doc](https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_conv), input value can be treated as unsigned if src_base is positive number, so we can't simply judge a value in range[INT64_MAX,UINT64_MAX] as overflow.

Reply:
@silverbullet233 Could we judge `dest_base_value > 0 && decimal_num > std::numeric_limits<uint64_t>::max() || dest_base_value < 0 && decimal_num > std::numeric_limits<int64_t>::max()`?